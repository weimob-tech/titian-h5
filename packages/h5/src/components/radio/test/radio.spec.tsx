import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { TiRadio } from '..';
import { TestContainer } from '../../common/test/container';

describe('渲染单选框', () => {
  it('基本渲染', async () => {
    const page = await newSpecPage({
      components: [TiRadio, TestContainer],
      template: () => (
        <test-container>
          <ti-radio />
          <ti-radio label="文字内容" />
          <ti-radio>文字内容</ti-radio>
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('默认选中状态', async () => {
    const page = await newSpecPage({
      components: [TiRadio],
      template: () => (
        <test-container>
          <ti-radio label="默认非受控模式" />
          <ti-radio label="默认选中，受控模式" checked />
          <ti-radio label="默认选中，非受控模式" default-checked />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('禁用点击效果', async () => {
    const page = await newSpecPage({
      components: [TiRadio, TestContainer],
      template: () => (
        <test-container>
          <ti-radio label="禁用全部点击效果" disabled />
          <ti-radio label="禁用文字点击效果" label-disabled />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
  it('修改单选框的圆角度数', async () => {
    const page = await newSpecPage({
      components: [TiRadio],
      template: () => (
        <test-container>
          <ti-radio label="方型" shape="square" />
          <ti-radio label="圆型" shape="circle" />
          <ti-radio label="自定义角度" shape={6} />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('组件图标设置', async () => {
    const page = await newSpecPage({
      components: [TiRadio, TestContainer],
      template: () => (
        <test-container>
          <ti-radio label="默认图标" />
          <ti-radio label="自定义图标大小" size={64} />
          <ti-radio label="自定义图标" icon="plus" />
          <ti-radio label="自定义图标为空" icon="" />
          <ti-radio label="自定义颜色" color="blue" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('组件图标设置', async () => {
    const page = await newSpecPage({
      components: [TiRadio],
      template: () => <ti-radio label="自定义图标 false" icon={false} />,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('监听点击切换事件', async () => {
    const change = jest.fn();
    const page = await newSpecPage({
      components: [TiRadio],
      template: () => <ti-radio label="单选框" onChange={change} />,
    });
    {
      // label 点击
      const label = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-radio-label');
      label.click();
      expect(change).toBeCalledTimes(1);
      expect(change.mock.lastCall[0].detail).toBeTruthy();
    }
    {
      // icon 点击
      const iconWrap = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-radio-icon-wrap');
      iconWrap.click();
      expect(change).toBeCalledTimes(2);
      expect(change.mock.lastCall[0].detail).toBeFalsy();
    }
  });

  it('禁用全部点击切换事件', async () => {
    const change = jest.fn();
    const page = await newSpecPage({
      components: [TiRadio],
      template: () => <ti-radio label="单选框" onChange={change} disabled />,
    });
    {
      // label 点击
      const label = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-radio-label');
      label.click();
      expect(change).toBeCalledTimes(0);
    }
    {
      // icon 点击
      const iconWrap = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-radio-icon-wrap');
      iconWrap.click();
      expect(change).toBeCalledTimes(0);
    }
  });

  it('禁用label点击切换事件', async () => {
    const change = jest.fn();
    const page = await newSpecPage({
      components: [TiRadio],
      template: () => <ti-radio label="单选框" onChange={change} labelDisabled />,
    });
    {
      // label 点击
      const label = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-radio-label');
      label.click();
      expect(change).toBeCalledTimes(0);
    }
    {
      // icon 点击
      const iconWrap = page.root.shadowRoot.querySelector<HTMLDivElement>('.titian-radio-icon-wrap');
      iconWrap.click();
      expect(change).toBeCalledTimes(1);
      expect(change.mock.lastCall[0].detail).toBeTruthy();
    }
  });
});
