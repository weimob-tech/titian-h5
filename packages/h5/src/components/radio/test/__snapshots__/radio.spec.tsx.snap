// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染单选框 修改单选框的圆角度数 1`] = `
<ti-radio>
  <mock:shadow-root>
    <div class="titian-radio titian-radio-horizontal">
      <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: -499.5px;">
        <div class="titian-radio-icon-wrap">
          <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
        </div>
      </div>
      <div aria-hidden="true" class="titian-radio- titian-radio-label">
        方型
      </div>
    </div>
  </mock:shadow-root>
</ti-radio>
`;

exports[`渲染单选框 基本渲染 1`] = `
<test-container>
  <!---->
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio-empty titian-radio-label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          文字内容
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio-empty titian-radio-label">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
    文字内容
  </ti-radio>
</test-container>
`;

exports[`渲染单选框 禁用点击效果 1`] = `
<test-container>
  <!---->
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-disabled titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          禁用全部点击效果
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio label-disabled="">
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          禁用文字点击效果
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
</test-container>
`;

exports[`渲染单选框 组件图标设置 1`] = `
<test-container>
  <!---->
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          默认图标
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="64"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          自定义图标大小
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="plus" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          自定义图标
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <slot name="icon"></slot>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          自定义图标为空
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
  <ti-radio>
    <mock:shadow-root>
      <div class="titian-radio titian-radio-horizontal">
        <div aria-hidden="true" class="titian-checkbox-icon-content" style="--radio-icon-color: blue; --base-radius-size: 499.5px;">
          <div class="titian-radio-icon-wrap">
            <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
          </div>
        </div>
        <div aria-hidden="true" class="titian-radio- titian-radio-label">
          自定义颜色
        </div>
      </div>
    </mock:shadow-root>
  </ti-radio>
</test-container>
`;

exports[`渲染单选框 组件图标设置 2`] = `
<ti-radio>
  <mock:shadow-root>
    <div class="titian-radio titian-radio-horizontal">
      <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
        <div class="titian-radio-icon-wrap">
          <slot name="icon"></slot>
        </div>
      </div>
      <div aria-hidden="true" class="titian-radio- titian-radio-label">
        自定义图标 false
      </div>
    </div>
  </mock:shadow-root>
</ti-radio>
`;

exports[`渲染单选框 默认选中状态 1`] = `
<ti-radio>
  <mock:shadow-root>
    <div class="titian-radio titian-radio-horizontal">
      <div aria-hidden="true" class="titian-checkbox-icon-content" style="--base-radius-size: 499.5px;">
        <div class="titian-radio-icon-wrap">
          <ti-icon ext-class="titian-radio-icon titian-radio-icon-zoom-out" name="checkbox-hollow" size="32"></ti-icon>
        </div>
      </div>
      <div aria-hidden="true" class="titian-radio- titian-radio-label">
        默认非受控模式
      </div>
    </div>
  </mock:shadow-root>
</ti-radio>
`;
